src/otx/cli/cli.py

```diff
@@ -0,0 +1,316 @@
+            description="OpenVINO Training-Extension command line tool",
+            env_prefix="otx",
+            formatter_class=CustomHelpFormatter,
+        )
+        parser.add_argument(
+            "-v",
+            "--version",
+            action="version",
+            version=f"%(prog)s {__version__}",
+            help="Display OTX version number.",
+        )
+        return parser
+
+    def subcommand_parser(self, **kwargs) -> ArgumentParser:
+        """Returns an ArgumentParser object for parsing command line arguments specific to a subcommand.
+
+        Returns:
+            ArgumentParser: An ArgumentParser object configured with the specified arguments.
+        """
+        parser = ArgumentParser(
+            formatter_class=CustomHelpFormatter,
+            **kwargs,
+        )
+        parser.add_argument(
+            "-v",
+            "--verbose",
+            action="count",
+            help="Verbose mode. This shows a configuration argument that allows for more specific overrides. \
+                Multiple -v options increase the verbosity. The maximum is 2.",
+        )
+        parser.add_argument(
+            "-c",
+            "--config",
+            action=ActionConfigFile,
+            help="Path to a configuration file in json or yaml format.",
+        )
+        parser.add_argument(
+            "--data_root",
+            type=str,
+            help="Path to dataset root.",
+        )
+        parser.add_argument(
+            "--task",
+            type=str,
+            help="Task Type.",
+        )
+        parser.add_argument(
+            "--callback_monitor",
+            type=str,
+            help="The metric for that model to monitor in the callback.",
+        )
+        return parser
+
+    @staticmethod
+    def engine_subcommands() -> dict[str, set[str]]:
+        """Returns a dictionary of engine subcommands and their required arguments.
+
+        Returns:
+            A dictionary where the keys are the subcommands and the values are sets of required arguments.
+        """
+        device_kwargs = {"accelerator", "devices"}
+        return {
+            "train": device_kwargs,
+            "test": {"datamodule"}.union(device_kwargs),
+            "predict": {"datamodule"}.union(device_kwargs),
+            "export": device_kwargs,
+        }
+
+    def add_subcommands(self) -> None:
+        """Adds subcommands to the CLI parser.
+
+        This method initializes and configures subcommands for the OTX CLI parser.
+        It iterates over the available subcommands, adds arguments specific to each subcommand,
+        and registers them with the parser.
+
+        Returns:
+            None
+        """
+        self._subcommand_parsers: dict[str, ArgumentParser] = {}
+        parser_subcommands = self.parser.add_subcommands()
+        self._set_extension_subcommands_parser(parser_subcommands)
+        if not _ENGINE_AVAILABLE:
+            return
```

Comment:
This will return if no environment is configured to use Engine (without torch or lightning), causing the otx subcommand to only show install. I'll add a comment here.